File: evals/fixtures/hotel-express-ts/app/src/models/store.ts
import { Guest, Room, Booking } from "./types";

let idCounter = 0;

export function generateId(): string {
  idCounter++;
  return `id-${Date.now()}-${idCounter}`;
}

export const store = {
  guests: [] as Guest[],
  rooms: [
    { id: "room-101", number: "101", floor: 1, type: "standard" },
    { id: "room-205", number: "205", floor: 2, type: "suite" },
    { id: "room-ph1", number: "PH1", floor: 3, type: "penthouse" },
  ] as Room[],
  bookings: [] as Booking[],
};

export function findRoom(id: string): Room | undefined {
  return store.rooms.find((r) => r.id === id);
}

export function findGuest(id: string): Guest | undefined {
  return store.guests.find((g) => g.id === id);
}

File: evals/fixtures/hotel-flask-py/app/models/store.py
import uuid

from models.booking import Booking, Guest
from models.room import Room

# In-memory data store
rooms = [
    Room(id="room-101", number="101", floor=1, type="standard"),
    Room(id="room-205", number="205", floor=2, type="suite"),
    Room(id="room-ph1", number="PH1", floor=3, type="penthouse"),
]

guests: list[Guest] = []

bookings: list[Booking] = []


def generate_id() -> str:
    return str(uuid.uuid4())


def find_room(room_id: str) -> Room | None:
    return next((r for r in rooms if r.id == room_id), None)


def find_guest(guest_id: str) -> Guest | None:
    return next((g for g in guests if g.id == guest_id), None)
